File: app/src/main/java/com/github/mbarrben/moviedb/MainActivity.java
package com.github.mbarrben.moviedb;


import com.github.mbarrben.moviedb.di.UIBus;
import com.github.mbarrben.moviedb.domain.GetMovies;
import com.github.mbarrben.moviedb.model.entities.Movie;
import com.squareup.otto.Bus;
import com.squareup.otto.Subscribe;

import android.os.Bundle;
import android.util.Log;

import java.util.LinkedList;
import java.util.List;

import javax.inject.Inject;

public class MainActivity extends BaseActivity {

    @Inject
    GetMovies getMovies;
    @Inject
    @UIBus
    Bus bus;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected List<Object> getModules() {
        LinkedList<Object> modules = new LinkedList<>();
        modules.add(new UIModule());
        return modules;
    }

    @Override
    protected void onResume() {
        super.onResume();
        bus.register(this);
        getMovies.get();
    }

    @Override
    protected void onPause() {
        super.onPause();
        bus.unregister(this);
    }

    @Subscribe
    public void onMoviesReceived(Movie.List movies) {
        Log.d("Movida", "onMoviesReceived");
    }
}

File: app/src/main/java/com/github/mbarrben/moviedb/app/MainActivity.kt
package com.github.mbarrben.moviedb.app

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.runtime.CompositionLocalProvider
import coil.ImageLoader
import com.github.mbarrben.moviedb.navigation.NavigationManager
import com.github.mbarrben.moviedb.navigation.composables.Navigation
import com.github.mbarrben.moviedb.ui.theme.MovieDbTheme
import com.google.accompanist.coil.LocalImageLoader
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class MainActivity : ComponentActivity() {

    @Inject
    lateinit var navigationManager: NavigationManager
    @Inject
    lateinit var imageLoader: ImageLoader

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            MovieDbTheme {
                CompositionLocalProvider(LocalImageLoader provides imageLoader) {
                    Navigation(navigationManager)
                }
            }
        }
    }
}
